File: bookly-parent/admin-backend/src/main/java/com/lgs/admin/backend/api/CategoryApi.java
package com.lgs.admin.backend.api;

import com.lgs.common.model.Category;
import com.lgs.common.service.CategoryService;
import org.apache.dubbo.config.annotation.DubboReference;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

import java.util.List;
import java.util.Map;

@RestController
@RequestMapping(value = "/api/v1/categories", produces = "application/json;charset=utf-8")
public class CategoryApi {
    @DubboReference
    private CategoryService categoryService;

    @GetMapping("/all")
    public Map<String, Object> findAll() {
        List<Category> categories = this.categoryService.findAll();
        return Map.of("success",true,"data", categories);
    }
    @GetMapping("/tree")
    public Map<String, Object> findTree() {
        Category category = this.categoryService.findTree();
        return Map.of("success",true,"data", category);
    }
    @GetMapping("/nameTree")
    public Map<String, Object> searchNameTree(Category category) {
        System.out.println(category);
        System.out.println(category.getName());
        Category tree = this.categoryService.searchNameTree(category);
        if(tree == null) {
            return Map.of("success",false,"data", "未找到该类别");
        }
        return Map.of("success",true,"data", tree);
    }
    @GetMapping("/{id}")
    public Map<String, Object> findById(@PathVariable("id") Integer id) {
        Category category = this.categoryService.findById(id);
        return Map.of("success",true,"data", category);
    }
    @GetMapping("/names")
    public Map<String, Object> findByNames() {
        List<Category> categories = this.categoryService.getCategoryNames();
        return Map.of("success",true,"data", categories);
    }
    @PostMapping
    public Map<String, Object> save(@RequestBody Category category) {
        boolean success = this.categoryService.save(category);
        return Map.of("success",success);
    }
    @PutMapping
    public Map<String, Object> update(@RequestBody Category category) {
        boolean success = this.categoryService.update(category);
        return Map.of("success",success);
    }
    @DeleteMapping("/{id}")
    public Map<String, Object> delete(@PathVariable("id") Integer id) {
        boolean success = this.categoryService.delete(id);
        return Map.of("success",success);
    }
}

File: admin-frontend/src/api/CategoryApi.js
import api from "../utils/api"
const baseUrl = "/categories"
async function findAll () {
    let resp = await api({
        url: baseUrl + "/tree",
        method: "get",
    })
    return resp.data.children;
}
async function apiFindNameTree (params) {
    let resp = await api({
        url: baseUrl + "/nameTree",
        method: "get",
        params: {
            ...params
        }
    })
    console.log("resp", resp)
    if (resp.success)
        return resp.data.children;
    else
        return [];
}
async function findById () {
    let resp = await api({
        url: baseUrl + `/${id}`,
        method: "get",
    })
    return resp;
}
async function findCategoryNames () {
    let resp = await api({
        url: baseUrl + "/names",
        method: "get",
    })
    return resp.data;
}
async function deleteById (id) {
    let resp = await api({
        url: baseUrl + `/${id}`,
        method: "delete",
    });
    return resp;
}
async function save (doctor) {

    let resp = await api({
        url: baseUrl,
        method: "post",
        data: doctor
    })
    return resp;
}
async function update (doctor) {
    console.log("doctor", doctor)
    let p = Object.assign({}, doctor)
    delete p.children;
    console.log("p", p)
    let resp = await api({
        url: baseUrl,
        method: "put",
        data: doctor
    })
    return resp;
}
export { findAll, deleteById, save, update, findById, apiFindNameTree, findCategoryNames }